### ISSUE TITLE
Cannot start scans since 1.2

### ISSUE THREAD

**Issue by AntoineBrunet** [KSPModStewards/SCANsat#244]
Playing KSP on Linux (Debian) with the latest release of SCANsat.
Since the 1.2 update, I can't seem to start any scan. When right-clicking the option does show up, but clicking on the button does nothing. There seem to be exception raised, from the log:

```
[EXC 17:05:00.422] NotSupportedException: The invoked member is not supported in a dynamic module.
System.Reflection.Emit.AssemblyBuilder.GetExportedTypes ()
SCANsat.SCAN_Toolbar.ToolbarTypes.getType (System.String name)
SCANsat.SCAN_Toolbar.ToolbarManager.get_Instance ()
SCANsat.SCAN_Toolbar.ToolbarManager.get_ToolbarAvailable ()
SCANsat.SCAN_PartModules.SCANsat.startScan ()
BaseEvent.Invoke ()
UIPartActionButton.OnClick ()
UnityEngine.Events.InvokableCall.Invoke (System.Object[] args)
UnityEngine.Events.InvokableCallList.Invoke (System.Object[] parameters)
UnityEngine.Events.UnityEventBase.Invoke (System.Object[] parameters)
UnityEngine.Events.UnityEvent.Invoke ()
UnityEngine.UI.Button.Press ()
UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData)
UnityEngine.EventSystems.ExecuteEvents.Execute (IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData)
UnityEngine.EventSystems.ExecuteEvents.Execute[IPointerClickHandler] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.EventFunction`1 functor)
UnityEngine.EventSystems.EventSystem:Update()
[EXC 17:05:05.596] NotSupportedException: The invoked member is not supported in a dynamic module.
System.Reflection.Emit.AssemblyBuilder.GetExportedTypes ()
SCANsat.SCAN_Toolbar.ToolbarTypes.getType (System.String name)
SCANsat.SCAN_Toolbar.ToolbarManager.get_Instance ()
SCANsat.SCAN_Toolbar.ToolbarManager.get_ToolbarAvailable ()
SCANsat.SCAN_PartModules.SCANsat.startScan ()
BaseEvent.Invoke ()
UIPartActionButton.OnClick ()
UnityEngine.Events.InvokableCall.Invoke (System.Object[] args)
UnityEngine.Events.InvokableCallList.Invoke (System.Object[] parameters)
UnityEngine.Events.UnityEventBase.Invoke (System.Object[] parameters)
UnityEngine.Events.UnityEvent.Invoke ()
UnityEngine.UI.Button.Press ()
UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData)
UnityEngine.EventSystems.ExecuteEvents.Execute (IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData)
UnityEngine.EventSystems.ExecuteEvents.Execute[IPointerClickHandler] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.EventFunction`1 functor)
UnityEngine.EventSystems.EventSystem:Update()
```

Attached is the full log.
[KSP_log.txt](https://github.com/S-C-A-N/SCANsat/files/531262/KSP_log.txt)

**Comment by DMagic1** [comment]
Are you using Blizzy's toolbar? SCANsat probably needs to be updated for that.

**Comment by AntoineBrunet** [comment]
I was not using Blizzy's toolbar. After installing the latest toolbar version (1.7.13), SCANsat started working.
Not sure if we should close this issue as installing Blizzy's toolbar seems to fix it, or it should remain open as SCANsat should work without it?
Thank you anyway!

**Comment by mcstalker** [comment]
In the VAB you can open the scanning with the extend/Retract button but when you are on the pad the scanners will not extend when you click on the start button.  I then set the Action keys one to start, one to stop and then one to toggle,  Start and Toggle did nothing but when I clicked on stop the panels popped open and then closed.

**Comment by MaceWe** [comment]
Can confirm that. When Contract Configurator, even latest 1.2 release, ScanSat parts only animate in VAB/SPH but refuse to work in the world instance

**Comment by MaceWe** [comment]
And canconfirm that installing latest Toolbar API fixes the issue

**Comment by ahmedcharles** [comment]
I made a pull request (#248) with the fix, which should make it slightly easier to get the fix into a release. :)

**Comment by DMagic1** [comment]
fixed in 16.10
